pkg/agent/core/ngt/handler/grpc/handler_test.go

```diff
@@ -3972,90 +3972,900 @@ func Test_server_MultiInsert(t *testing.T) {
+				return errors.Errorf("got uuid: \"%#v\",\n\t\t\t\twant uuid: \"%#v\"", uuid, w.wantUuid)
+			}
 		}
 		return nil
 	}
+
+	const (
+		insertNum = 1000
+		dimension = 128
+	)
+
+	defaultUpdateConfig := &payload.Update_Config{
+		SkipStrictExistCheck: true,
+	}
+	toFloat32Vector := func(v []uint8) (ret []float32) {
+		ret = make([]float32, 0, len(v))
+		for _, e := range v {
+			ret = append(ret, float32(e))
+		}
+		return
+	}
+	beforeFunc := func(objectType string) func(args) (Server, error) {
+		cfg := &config.NGT{
+			Dimension:        dimension,
+			DistanceType:     ngt.L2.String(),
+			ObjectType:       objectType,
+			CreationEdgeSize: 60,
+			SearchEdgeSize:   20,
+			KVSDB: &config.KVSDB{
+				Concurrency: 10,
+			},
+			VQueue: &config.VQueue{
+				InsertBufferPoolSize: 1000,
+				DeleteBufferPoolSize: 1000,
+			},
+		}
+		var gen func(int, int) [][]float32
+		switch objectType {
+		case ngt.Float.String():
+			gen = vector.GaussianDistributedFloat32VectorGenerator
+		case ngt.Uint8.String():
+			gen = func(n, dim int) (ret [][]float32) {
+				vectors := vector.GaussianDistributedUint8VectorGenerator(n, dim)
+				ret = make([][]float32, 0, len(vectors))
+				for _, vector := range vectors {
+					ret = append(ret, toFloat32Vector(vector))
+				}
+				return
+			}
+		}
+
+		return func(a args) (Server, error) {
+			eg, ctx := errgroup.New(a.ctx)
+			ngt, err := service.New(cfg, service.WithErrGroup(eg), service.WithEnableInMemoryMode(true))
+			if err != nil {
+				return nil, err
+			}
+
+			s, err := New(WithErrGroup(eg), WithNGT(ngt))
+			if err != nil {
+				return nil, err
+			}
+
+			reqs := make([]*payload.Insert_Request, insertNum)
+			for i, v := range gen(insertNum, cfg.Dimension) {
+				reqs[i] = &payload.Insert_Request{
+					Vector: &payload.Object_Vector{
+						Id:     strconv.Itoa(i),
+						Vector: v,
+					},
+					Config: &payload.Insert_Config{
+						SkipStrictExistCheck: true,
+					},
+				}
+			}
+			reqs[0].Vector.Id = a.indexId
+			if a.indexVector != nil {
+				reqs[0].Vector.Vector = a.indexVector
+			}
+			if _, err := s.MultiInsert(ctx, &payload.Insert_MultiRequest{Requests: reqs}); err != nil {
+				return nil, err
+			}
+			if _, err := s.CreateIndex(ctx, &payload.Control_CreateIndexRequest{PoolSize: 100}); err != nil {
+				return nil, err
+			}
+			return s, nil
+		}
+	}
+
+	utf8ToSjis := func(s string) string {
+		b, _ := ioutil.ReadAll(transform.NewReader(strings.NewReader(s), japanese.ShiftJIS.NewEncoder()))
+		return string(b)
+	}
+
+	utf8ToEucjp := func(s string) string {
+		b, _ := ioutil.ReadAll(transform.NewReader(strings.NewReader(s), japanese.EUCJP.NewEncoder()))
+		return string(b)
+	}
+
+	fillFloat32 := func(val float32, size int) (ret []float32) {
+		ret = make([]float32, size)
+		for i := range ret {
+			ret[i] = val
+		}
+		return
+	}
+	fillUint8 := func(val uint8, size int) (ret []uint8) {
+		ret = make([]uint8, size)
+		for i := range ret {
+			ret[i] = val
+		}
+		return
+	}
```

Comment:
How about defining these as global functions in this test file?

Reply:
I agree with you.
In my opinion, these functions will be usable as utility functions for other handlers.
What do you think about implementing test utility package like pkg/testing?

Reply:
Sorry, I rethink that they implement to file global in this PR.
And if we have time, we will refactor them.